src/Microsoft.AspNet.OData.Shared/DeltaSetOfT.cs

```diff
@@ -0,0 +1,49 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="DeltaSetOfT.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved. 
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using Microsoft.AspNet.OData.Common;
+
+namespace Microsoft.AspNet.OData
+{
+    /// <summary>
+    /// Represents an <see cref="IDeltaSet"/> that is a collection of <see cref="IDeltaSetItem"/>s.
+    /// </summary>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Naming", "CA1710:IdentifiersShouldHaveCorrectSuffix")]
+    [NonValidatingParameterBinding]
+    public class DeltaSet<TStructuralType> : Collection<IDeltaSetItem>, IDeltaSet where TStructuralType : class
+    {
+        private Type _clrType;
+        private IList<string> _keys;
```

Comment:
`_keys_ is a private field that is not exposed via a public or protected property anywhere. I also don't see it being applied anywhere. What purpose does it serve?

Reply:
We use _keys when creating a deltaset during deserialization. Also there are some methods that will be added later in this class that use this field.